test/test_google_cloud_manage.py

```diff
@@ -652,34 +656,65 @@ def test_get_primary_service_account(test_cloud_manager):
     """
     # Setup #
     test_domain = "test-domain.net"
-    primary_service_account = "primary-account" + test_domain
-
     new_member_1_id = "1"
-    new_member_1_email = new_member_1_id + "@" + test_domain
-    group_id = new_member_1_id + "-testuser"  # This has to be the user's ID - Username
+    new_member_1_username = "testuser"
+    primary_service_account = _get_proxy_group_service_account_id_for_user(
+        new_member_1_id, new_member_1_username
+    ) + "@" + test_domain
+
+    group_name = _get_proxy_group_name_for_user(
+        new_member_1_id, new_member_1_username)
+    group_email = group_name + "@" + test_domain
+
+    test_cloud_manager.get_group = MagicMock()
+    test_cloud_manager.get_group.return_value = {
+        "kind": "admin#directory#group",
+        "id": group_name,
+        "etag": "",
+        "email": group_name + "@" + test_domain,
+        "name": "",
+        "directMembersCount": 0,
+        "description": "",
+        "adminCreated": False,
+        "aliases": [
+            ""
+        ],
+        "nonEditableAliases": [
+            ""
+        ]
+    }
 
     test_cloud_manager.get_service_accounts_from_group = MagicMock()
-    test_cloud_manager.get_service_accounts_from_group.return_value = [new_member_1_email]
+    test_cloud_manager.get_service_accounts_from_group.return_value = [primary_service_account]
+
     test_cloud_manager.get_service_account = MagicMock()
-    test_cloud_manager.get_service_account.return_value = primary_service_account
-    test_cloud_manager._service_account_email_domain = test_domain
+    test_cloud_manager.get_service_account.return_value = {
+        "name": "",
+        "projectId": "",
+        "uniqueId": "",
+        "email": primary_service_account,
+        "displayName": "",
+        "etag": "",
+        "oauth2ClientId": "",
+    }
 
     # Call #
-    email = test_cloud_manager.get_primary_service_account(group_id)
+    response = test_cloud_manager.get_primary_service_account(group_name)
+    email = response["email"]
 
     # Test #
-    assert email == primary_service_account
+    assert email == group_email
```

Comment:
It might make sense to move the fixture up until here into a `conftest.py`, and then put these asserts into their own test without the fixture.

Reply:
Yeah, there's a lot of stuff that could be cleaned up in these tests with fixtures... I'll take a look

Reply:
attempted to clean stuff up in f4feb25